spyder/plugins/editor/widgets/codeeditor/multicursor_mixin.py

```diff
@@ -0,0 +1,521 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Mixin to manage editing with multiple cursors.
+"""
+
+# Standard library imports
+import functools
+import itertools
+
+# Third party imports
+from qtpy.QtCore import Qt, QTimer, Slot
+from qtpy.QtGui import (
+    QFontMetrics, QPaintEvent, QPainter, QTextCursor, QKeyEvent
+)
+from qtpy.QtWidgets import QApplication
+
+# Local imports
+from spyder.plugins.editor.api.decoration import TextDecoration
+
+
+class MultiCursorMixin:
+    """Mixin to manage editing with multiple cursors."""
+
+    def init_multi_cursor(self):
+        """Initialize attrs and callbacks for multi-cursor functionality"""
+        # actual default comes from setup_editor default args
+        self.multi_cursor_enabled = False
+        self.cursor_width = self.get_conf('cursor/width', section='main')
+        self.overwrite_mode = self.overwriteMode()
+
+        # Track overwrite manually when for painting reasons with multi-cursor
+        self.setOverwriteMode(False)
+        self.setCursorWidth(0)  # draw our own cursor
+        self.extra_cursors = []
+        self.cursor_blink_state = False
+        self.cursor_blink_timer = QTimer(self)
+        self.cursor_blink_timer.setInterval(
+            QApplication.cursorFlashTime() // 2
+        )
+        self.cursor_blink_timer.timeout.connect(
+            self._on_cursor_blinktimer_timeout
+        )
+        self.focus_in.connect(self.start_cursor_blink)
+        self.focus_changed.connect(self.stop_cursor_blink)
+        self.painted.connect(self.paint_cursors)
+        self.multi_cursor_ignore_history = False
+        self._drag_cursor = None
+
+    def __selection_colors(self, cache=[]):
```

Comment:
Instead of using a mutable kwarg, which is not a good idea (see PR #23293), please use a class attribute called `self.__cached_colors`. You can initialize it to `None` and then set it here.

Reply:
Resolved by going back to `SpyderPalette` instead of pulling `widget.palette()`